ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java

```diff
@@ -0,0 +1,1161 @@
+    // Given:
+    givenUnwindowedSourceStream();
+
+    // When:
+    final KStreamHolder<?> builtKstream = streamSource.build(planBuilder, planInfo);
+
+    // Then:
+    assertThat(builtKstream.getSchema(), is(SCHEMA_WITH_NEW_PSEUDOCOLUMNS));
+  }
+
+  @Test
+  public void shouldReturnCorrectSchemaForLegacyUnwindowedSourceStream() {
+    // Given:
+    givenUnwindowedSourceStream(LEGACY_PSEUDOCOLUMN_VERSION_NUMBER);
+
+    // When:
+    final KStreamHolder<?> builtKstream = streamSource.build(planBuilder, planInfo);
+
+    // Then:
+    assertThat(builtKstream.getSchema(), is(SCHEMA_WITHOUT_NEW_PSEUDOCOLUMNS));
+  }
+
+  @Test
+  public void shouldReturnCorrectSchemaForUnwindowedSourceTable() {
+    // Given:
+    givenUnwindowedSourceTableV1(true);
+
+    // When:
+    final KTableHolder<GenericKey> builtKTable = tableSourceV1.build(planBuilder, planInfo);
+
+    // Then:
+    assertThat(builtKTable.getSchema(), is(SCHEMA_WITHOUT_NEW_PSEUDOCOLUMNS));
+  }
+
+  @Test
+  public void shouldThrowWhenTryingToCreateUnwindowedTableWithNewPseudoColumnVersion() {
+    // Given:
+    givenUnwindowedSourceTableV1(true, ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION);
+
+    // When:
+    final Exception e = assertThrows(
+        KsqlException.class,
+        () -> tableSourceV1.build(planBuilder, planInfo)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Should not be using old execution step version with new queries"));
+  }
+
+  @Test
+  public void shouldBuildSourceValueSerdeCorrectly() {
+    // Given:
+    givenUnwindowedSourceStream();
+
+    // When:
+    streamSource.build(planBuilder, planInfo);
+
+    // Then:
+    verify(buildContext).buildValueSerde(valueFormatInfo, PHYSICAL_SCHEMA, ctx);
+  }
+
+  @Test
+  public void shouldBuildSourceKeySerdeCorrectly() {
+    // Given:
+    givenWindowedSourceStream();
+
+    // When:
+    windowedStreamSource.build(planBuilder, planInfo);
+
+    // Then:
+    verify(buildContext).buildKeySerde(
+        keyFormatInfo,
+        windowInfo,
+        PhysicalSchema.from(SOURCE_SCHEMA, KEY_FEATURES, VALUE_FEATURES),
+        ctx
+    );
+  }
+
+  @Test
+  public void shouldReturnCorrectSchemaForWindowedSourceStream() {
+    // Given:
+    givenWindowedSourceStream();
+
+    // When:
+    final KStreamHolder<?> builtKstream = windowedStreamSource.build(planBuilder, planInfo);
+
+    // Then:
+    assertThat(builtKstream.getSchema(), is(WINDOWED_SCHEMA_WITH_NEW_PSEUDOCOLUMNS));
+  }
+
+  @Test
+  public void shouldReturnCorrectSchemaForLegacyWindowedSourceStream() {
+    // Given:
+    givenWindowedSourceStream(LEGACY_PSEUDOCOLUMN_VERSION_NUMBER);
+
+    // When:
+    final KStreamHolder<?> builtKstream = windowedStreamSource.build(planBuilder, planInfo);
+
+    // Then:
+    assertThat(builtKstream.getSchema(), is(WINDOWED_SCHEMA_WITHOUT_NEW_PSEUDOCOLUMNS));
+  }
+
+  @Test
+  public void shouldReturnCorrectSchemaForWindowedSourceTable() {
```

Comment:
The name of this test does not match the behavior of the test.

Reply:
Also, is this test identical to `shouldThrowOnTryToGetTransformerFromIncompatibleTableSource()` below?

Reply:
More or less. The other is testing that the `getTransformerFromTableSource()` method will throw while this is testing that the `build()` method will throw. In essence, these are the same. My mindset was that since the old testing suite contained both of these tests, refactoring the tests to check for the new expected behavior being accurate was better than removing them. If you think this is redundant, I don't disagree and can remove them.

Reply:
I see, thanks for calling out the difference. I think they are redundant. The first line of getTransformerFromTableSource() calls build(), which is the public method from SourceBuilderV1 that we are actually interested in testing.

Let's remove `shouldThrowOnTryToGetTransformerFromIncompatibleTableSource()` and rename this one (for testing `build()`) to be named appropriately.